monika/linux/generic_syscall.cpp

```diff
@@ -0,0 +1,58 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include "debugbreak.h"
+#include <string.h>
+#include <haiku_errors.h>
+#include <BeDefs.h>
+#include "linux_syscall.h"
+
+#define RANDOM_SYSCALLS "random"
+#define RANDOM_GET_ENTROPY 1
+struct random_get_entropy_args {
+	void *buffer;
+	size_t length;
+};
+static inline int _open(const char *hostPath, int mode) {
+	return LINUX_SYSCALL4(__NR_openat, AT_FDCWD, hostPath, 0, mode);
+}
+static inline ssize_t _read(int fd, void *buffer, size_t bufferSize) {
+	return LINUX_SYSCALL3(__NR_read, fd, buffer, bufferSize);
+}
+static inline int _close(int fd) {
+        return LINUX_SYSCALL1(__NR_close, fd);
+}
+extern "C" {
+int32_t _moni_generic_syscall(const char *userSubsystem, uint32_t function, void *buffer, uint64_t bufferSize) {
```

Comment:
From what I understand, `_kern_generic_syscall` is a mechanism for Haiku kernel drivers to supply more system calls.

We should mimic this mechanism in `hyclone_server`. It's better to add a [servercall](https://github.com/trungnt2910/hyclone/blob/6a7ef2f557d87b7906c485a2d5773aff8f607d31/shared_headers/servercall_defs.h). Then, we can provide the `random` subsystem as a shared library that gets `dlopen`ed by `hyclone_server`.

It might be even nicer if we could emulate some kernel APIs and load Haiku's original subsystem driver.